dvc/repo/reproduce.py

```diff
@@ -155,20 +165,27 @@ def _reproduce_stages(
     else:
         pipeline = nx.dfs_postorder_nodes(G, node)
 
-    result = []
-    for n in pipeline:
-        try:
-            ret = _reproduce_stage(stages, n, **kwargs)
-
-            if len(ret) != 0 and ignore_build_cache:
-                # NOTE: we are walking our pipeline from the top to the
-                # bottom. If one stage is changed, it will be reproduced,
-                # which tells us that we should force reproducing all of
-                # the other stages down below, even if their direct
-                # dependencies didn't change.
-                kwargs["force"] = True
-
-            result += ret
-        except Exception as ex:
-            raise ReproductionError(stages[n].relpath, ex)
-    return result
+    pipeline = [stages[node] for node in pipeline]
+
+    try:
+        _lock(pipeline)
```

Comment:
Where do we handle lock error? what is the message? Does it state particular stage file that is busy?